### PR TITLE
fix: use dom builder for gutter tooltip and inline widget

### PR DESCRIPTION
**Pull request by akoreman**
*Issue #, if available:* #5575

*Description of changes:* Changes how we construct the gutter tooltip and inline widget to use the dom building utilities.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Pull Request Checklist:
* [x] No backwards incompatible changes were made to Ace's public interface which is defined through the main typings file (`ace.d.ts`) and its references:
    * https://github.com/ajaxorg/ace/blob/master/ace.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-modes.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-extensions.d.ts

### CODE REVIEW

**Review by whazor — APPROVED** [review]

File: src/virtual_renderer_test.js [hunk 1]
```diff
@@ -395,7 +395,7 @@ module.exports = {
         editor.renderer.$loop._flush();
         assert.equal(editor.renderer.content.textContent, "abcdefThis is a long test text that is longer than ");
 
-        assert.equal(editor.session.lineWidgets[0].el.innerHTML, "<div class=\"ghost_text_line_wrapped\">30 characters</div><div></div><div>Ghost3</div>");
+        assert.equal(editor.session.lineWidgets[0].el.innerHTML, `<div class="ghost_text_line_wrapped">30 characters</div><div class=""></div><div class="">Ghost3</div>`);
```
Comment (andredcoliveira) [inline src/virtual_renderer_test.js:398]: Where do the empty classes come from? Are they a consequence of using `dom.createElement`?

Comment (andredcoliveira) [reply src/virtual_renderer_test.js:398]: Could you confirm whether this was tested manually (e.g., in the kitchen sink) too?

**Review by andredcoliveira — APPROVED** [review]
Nice

File: src/virtual_renderer_test.js (hunk 1 above)
Comment (akoreman) [reply src/virtual_renderer_test.js:398]: the empty classes are a consequence from ` chunkDiv.className = el.wrapped ? "ghost_text_line_wrapped": "";`. I can change it so we only set the classname if it's needed that might be cleaner indeed.

Comment (akoreman) [reply src/virtual_renderer_test.js]: > Could you confirm whether this was tested manually (e.g., in the kitchen sink) too?

yes 

Screenshot 2024-06-27 at 11 22 21

**Review by andredcoliveira — APPROVED** [review]

**Pull request merged**